src/modules/Game.class.js

```diff
@@ -20,25 +20,224 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+
+  moveLeft() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let i = 0; i < this.size; i++) {
+      const row = this.board[i].filter((val) => val !== 0); // тільки ненульові
+      const newRow = [];
+
+      for (let j = 0; j < row.length; j++) {
+        // Перевірка на об'єднання
+        if (row[j] === row[j + 1]) {
+          newRow.push(row[j] * 2);
+          this.score += row[j] * 2;
+          j++; // пропускаємо наступне число
+        } else {
+          newRow.push(row[j]);
+        }
+      }
+
+      while (newRow.length < this.size) {
+        newRow.push(0);
+      }
+
+      if (!this.arraysEqual(this.board[i], newRow)) {
+        this.board[i] = newRow;
+        moved = true;
+      }
+    }
+
+    if (moved) {
+      this.checkWin();
+      this.addRandomTile();
+      this.checkLose();
+    }
+
+    return moved;
+  }
+  arraysEqual(arr1, arr2) {
+    if (arr1.length !== arr2.length) {
+      return false;
+    }
+
+    return arr1.every((val, i) => val === arr2[i]);
+  }
+
+  moveRight() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    let moved = false;
+
+    for (let i = 0; i < this.size; i++) {
+      // 🔁 Реверс рядка
+      const row = [...this.board[i]].reverse().filter((val) => val !== 0);
+      const newRow = [];
+
+      for (let j = 0; j < row.length; j++) {
+        if (row[j] === row[j + 1]) {
+          newRow.push(row[j] * 2);
+          this.score += row[j] * 2;
+          j++;
+          moved = true;
+        } else {
+          newRow.push(row[j]);
+
+          if (this.board[i][this.size - 1 - j] !== row[j]) {
+            moved = true;
+          }
+        }
+      }
+
+      while (newRow.length < this.size) {
+        newRow.push(0);
+      }
+
```

Comment:
This logic is similar in every method, try to put it in a helper function and use it everywhere.